### DISCUSSION TITLE
c implementation of the internal stack management and record dispatching

### DISCUSSION THREAD

**Discussion by ibonelli** [getlogbook/logbook#390] [Ideas]
Hi there! In the documentation I found you plan to add a [c implementation of the internal stack management and record dispatching](https://logbook.readthedocs.io/en/stable/features.html#roadmap). I have C lang experience ([I've been teaching](https://info.nachodigital.com.ar/doku.php?id=start) it for ages) I've using Python for a good while now (among other things [I wrote my thesis in Python](https://github.com/ibonelli/pathplanning)) and I would love to help with a Python project like this one.

**Comment by RazerM** [comment]
This is not a priority right now, our benchmarks as is aren't particularly rigorous last I looked. I also only want to maintain Cython or Rust via PyO3.

The main issue facing logbook right now are #370, which when I investigated I discovered that the contextbound implementation is incorrect. (I do not use logbook with async so far)
